src/libxtp/anderson_mixing.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <boost/format.hpp>
+#include <votca/xtp/anderson_mixing.h>
+
+namespace votca {
+namespace xtp {
+
+void ANDERSON::UpdateOutput(const Eigen::VectorXd &newOutput) {
+
+  // Check if max mixing history is reached and adding new step to history
+  Index size = _output.size();
+  if (size > _max_history - 1) {
+    _output.erase(_output.begin());
+  }
+  _output.push_back(newOutput);
+}
+
+void ANDERSON::UpdateInput(const Eigen::VectorXd &newInput) {
+  Index size = _output.size();
+  if (size > _max_history - 1) {
+    _input.erase(_input.begin());
+  }
+  _input.push_back(newInput);
+}
+
+Eigen::VectorXd ANDERSON::NPAndersonMixing(const double alpha) {
+
+  _iteration++;
```

Comment:
can you not get the iteration cound from the size of input and output? Do you need an iteration counter?

Reply:
That should work as well.